### PR TITLE
Add support for bzip2 compression method

### LINKED ISSUE #22428: General compression methods: support for bz2.
**Use case**
Import and export of `bz2` files directly.

**Additional context**
It's extremely slow. But the use case is when there is already files in bz2 format and you have to deal with it.

### PR DESCRIPTION
**Pull request by evillique**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add support for `bzip2` compression method for import/export. Closes #22428

### CODE REVIEW

**Comment by evillique** [comment]
Should we add [bzip2](https://gitlab.com/federicomenaquintero/bzip2) library to ClickHouse-Extras? It seems like style-check does not allow repositories from GitLab.

**Comment by alexey-milovidov** [comment]
Yes if there is no GitHub mirror already.

**Comment by alexey-milovidov** [comment]
https://github.com/ClickHouse-Extras/bzip2

File: src/Common/ErrorCodes.cpp [hunk 1]
```diff
@@ -567,6 +567,8 @@
     M(1000, POCO_EXCEPTION) \
     M(1001, STD_EXCEPTION) \
     M(1002, UNKNOWN_EXCEPTION) \
+    M(1003, BZIP2_STREAM_DECODER_FAILED) \
```
Comment (alexey-milovidov) [inline src/Common/ErrorCodes.cpp:570]: Add it after other specific errors, and 1000+ is for generic errors.

**Review by alexey-milovidov — COMMENTED** [review]
Looks like everything else is perfect :+1:

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**